src/colmap/sfm/incremental_mapper.cc

```diff
@@ -839,6 +852,11 @@ std::shared_ptr<class Reconstruction> IncrementalMapper::Reconstruction()
   return reconstruction_;
 }
 
+std::shared_ptr<class ObservationManager>
```

Comment:
Didn't check all the callsites, but could we instead return a reference here instead of the shared_ptr or alternatively a const ref to the shared_ptr, otherwise each access to the shared_ptr incurs a copy and then a ref counter decrement when that object is destructred.

Reply:
Done.